ContactsManagerBackend/ContactsManager/ContactsManagerDAL/ContactsProvider.cs

```diff
@@ -38,5 +33,133 @@ public List<Contact> GetContacts()
 
             return Contacts;
         }
+
+        public Contact GetContactById(Guid Id)
+        {
+            var contact = new Contact();
+
+            using (SqlConnection conn = DBConnection.GetSqlConnection())
+            using (SqlCommand cmd = conn.CreateCommand())
+            {
+                cmd.CommandText = "getContactById";
+                cmd.CommandType = System.Data.CommandType.StoredProcedure;
+
+                var idParameter = new SqlParameter("Id", System.Data.SqlDbType.UniqueIdentifier);
+                idParameter.Value = Id;
+                cmd.Parameters.Add(idParameter);
+
+                var reader = cmd.ExecuteReader(System.Data.CommandBehavior.CloseConnection);
+
+                if (reader.Read())
+                {
+                    contact.LoadDataFromReader(reader);
+                }
+                else
+                {
+                    contact = null;
+                }
+            }
+
+            return contact;
+        }
+
+        public Contact CreateContact(Contact contact)
+        {
+            var createdContact = new Contact();
+            contact.Id = Guid.NewGuid();
+
+            using (SqlConnection conn = DBConnection.GetSqlConnection())
+            using (SqlCommand cmd = conn.CreateCommand())
+            {
+                cmd.CommandText = "createContact";
+                cmd.CommandType = System.Data.CommandType.StoredProcedure;
+
+                var idParam = new SqlParameter("Id", System.Data.SqlDbType.UniqueIdentifier);
```

Comment:
It would be less verbose to use an Add() overload of the Parameters property type. This applies to the whole file.
Also, please import System.Data namespace, rather than writing it everywhere